core/src/test/java/io/parsingdata/metal/ToStringTest.java

```diff
@@ -147,28 +151,39 @@ public void encoding() {
     @Test
     public void data() {
         final Environment environment = stream(1, 2);
-        final String envString = "stream: InMemoryByteStream(2); offset: 0; order: graph(EMPTY); callbacks: ";
+        final String envString = "source: InMemoryByteStream(2); offset: 0; order: graph(EMPTY); callbacks: ";
         assertEquals(envString, environment.toString());
         final ParseResult result = new ParseResult(true, environment);
         assertEquals("ParseResult(true, " + environment + ")", result.toString());
-        final ParseValue pv1 = new ParseValue("name", NONE, 0, new byte[]{1, 2}, enc());
+        final ParseValue pv1 = new ParseValue("name", NONE, bytesToSlice(new byte[]{1, 2}), enc());
         final String pv1String = "name(0x0102)";
         final OptionalValue ov1 = OptionalValue.of(pv1);
-        final OptionalValue ov2 = OptionalValue.of(new Value(new byte[]{3}, enc()));
+        final OptionalValue ov2 = OptionalValue.of(new Value(bytesToSlice(new byte[]{3}), enc()));
         assertEquals(">OptionalValue(0x03)>OptionalValue(" + pv1String + ")", ImmutableList.create(ov1).add(ov2).toString());
-        final ParseValue pv2 = new ParseValue("two", NONE, 0, new byte[]{3, 4}, enc());
+        final ParseValue pv2 = new ParseValue("two", NONE, bytesToSlice(new byte[]{3, 4}), enc());
         final String pv2String = "two(0x0304)";
         assertEquals(">" + pv2String + ">" + pv1String, ImmutableList.create(pv1).add(pv2).toString());
         assertEquals(">" + pv2String + ">" + pv1String, ImmutableList.create(pv1).add(pv2).toString());
     }
 
     @Test
+    public void slice() throws IOException {
+        final ParseValue pv1 = new ParseValue("name", NONE, bytesToSlice(new byte[]{1, 2}), enc());
+        assertEquals("0102@0:2", pv1.slice.toString());
```

Comment:
Agreed. Let's put an issue on the roadmap to come up with a good general rule for `toString()` and implement it everywhere :)

See #132.